nncf/torch/checkpoint_loading.py

```diff
@@ -18,7 +18,60 @@
 import torch
 
 from nncf.common.utils.logger import logger as nncf_logger
-from nncf.torch.utils import maybe_convert_legacy_names_in_model_state
+from nncf.common.compression import BaseCompressionAlgorithmController
+
+
+class PTCompressionStateVersion:
+    SAVE_NAME = 'version'
+    CURR_VERSION = '1.1'
+    COMPRESSION_STATE_STATE_DICT = '0.0'
+
+    def __init__(self, version):
+        self.version = version
+
+    @classmethod
+    def from_compression_state(cls, compression_state):
+        if cls.SAVE_NAME in compression_state:
+            return compression_state.get(cls.SAVE_NAME)
+        elif BaseCompressionAlgorithmController.BUILDER_STATE not in compression_state and \
+                BaseCompressionAlgorithmController.CONTROLLER_STATE not in compression_state:
+            return cls('0.0')
+        return cls('0.1')
+
+    def __eq__(self, other):
+        if self.version == other.version:
+            return True
+        return False
+
+    def __ne__(self, other):
+        if self.version != other.version:
+            return True
+        return False
+
+    def __lt__(self, other):
+        if self.version < other.version:
+            return True
+        return False
+
+    def __le__(self, other):
+        if self.version <= other.version:
+            return True
+        return False
+
+    def __gt__(self, other):
+        if self.version > other.version:
+            return True
+        return False
+
+    def __ge__(self, other):
+        if self.version >= other.version:
+            return True
+        return False
+
+    def is_state_dict(self):
+        if self == PTCompressionStateVersion(PTCompressionStateVersion.COMPRESSION_STATE_STATE_DICT):
```

Comment:
Version class should be very basic without auxiliary methods. do it outside of this class